test/run.sh

```diff
@@ -84,53 +84,52 @@ args+=( "-timeout=60m" )
 args+=( "-p=2" )
 
 if [[ "${GITPOD_REPO_ROOT:-}" != "" ]]; then
-  echo "Running in Gitpod workspace. Fetching USERNAME and USER_TOKEN" | werft log slice "test-setup"
+  echo "Running in Gitpod workspace. Fetching USERNAME and USER_TOKEN"
   USERNAME="$(kubectl --context=dev -n werft get secret integration-test-user -o jsonpath='{.data.username}' | base64 -d)"
   USER_TOKEN="$(kubectl --context=dev -n werft get secret integration-test-user -o jsonpath='{.data.token}' | base64 -d)"
   export USER_TOKEN
 else
-  echo "Running in Werft. Using INTEGRATION_TEST_USERNAME and INTEGRATION_TEST_USER_TOKEN for USERNAME and USER_TOKEN"  | werft log slice "test-setup"
+  echo "Using INTEGRATION_TEST_USERNAME and INTEGRATION_TEST_USER_TOKEN for USERNAME and USER_TOKEN"
   USERNAME="${INTEGRATION_TEST_USERNAME}"
   USER_TOKEN="${INTEGRATION_TEST_USER_TOKEN}"
   export USERNAME
   export USER_TOKEN
 fi
-werft log slice "test-setup" --done
 
 [[ "$USERNAME" != "" ]] && args+=( "-username=$USERNAME" )
 
+go install github.com/jstemmer/go-junit-report/v2@latest
+
+if ! npm list -g xunit-viewer; then npm install -g xunit-viewer; fi
+
+mkdir -p results
+
 if [ "$TEST_SUITE" == "workspace" ]; then
   TEST_NAME="workspace"
   LOG_FILE="${LOGS_DIR}/${TEST_NAME}.log"
 
   cd "$THIS_DIR"
-  echo "running integration for ${TEST_NAME} - log file at ${LOG_FILE}" | werft log slice "test-${TEST_NAME}-parallel"
+  echo "running integration for ${TEST_NAME}-parallel"
 
   set +e
   # shellcheck disable=SC2086
-  go test -v $TEST_LIST "${args[@]}" -run '.*[^.SerialOnly]$' 2>&1 | tee "${LOG_FILE}" | werft log slice "test-${TEST_NAME}-parallel"
+  go test -p 1 -v $TEST_LIST "${args[@]}" -run '.*[^.SerialOnly]$' 2>&1  | go-junit-report -subtest-mode=exclude-parents -set-exit-code -out "results/TEST-${TEST_NAME}-SERIAL.xml" -iocopy
   RC=${PIPESTATUS[0]}
   set -e
 
   if [ "${RC}" -ne "0" ]; then
     FAILURE_COUNT=$((FAILURE_COUNT+1))
-    werft log slice "test-${TEST_NAME}-parallel" --fail "${RC}"
-  else
-    werft log slice "test-${TEST_NAME}-parallel" --done
   fi
 
-  echo "running integration for ${TEST_NAME} - log file at ${LOG_FILE}" | werft log slice "test-${TEST_NAME}-serial-only"
+  echo "running integration for ${TEST_NAME}-serial-only"
   set +e
   # shellcheck disable=SC2086
-  go test -v $TEST_LIST "${args[@]}" -run '.*SerialOnly$' -p 1 2>&1 | tee -a "${LOG_FILE}" | werft log slice "test-${TEST_NAME}-serial-only"
+  go test -p 1 --parallel 1 -v $TEST_LIST "${args[@]}" -run '.*SerialOnly$' -p 1 2>&1 | go-junit-report -subtest-mode=exclude-parents -set-exit-code -out "results/TEST-${TEST_NAME}-PARALLEL.xml" -iocopy
```

Comment:
Updated to match params we're using with our Github Action workflow for workspace tests